src/commands.rs

```diff
@@ -0,0 +1,279 @@
+                            slot.to_string(),
+                            format!("{}", slot),
+                        )]
+                    })
+                    .collect::<Vec<_>>();
+
+                bot.send_message(chat_id, "Select the date and time:")
+                    .reply_markup(InlineKeyboardMarkup::new(buttons))
+                    .await?;
+
+                dialogue
+                    .update(State::ReceiveDateTime {
+                        user_name,
+                        service_name: service_name.to_string(),
+                    })
+                    .await?;
+            } else {
+                log::error!("Callback data is in an incorrect format: {}", callback_data);
+            }
+        } else {
+            log::error!("Message not found in callback query");
+        }
+    } else {
+        log::error!("Callback data not found");
+    }
+
+    Ok(())
+}
+
+async fn receive_data_time(bot: Bot, dialogue: MyDialogue, q: CallbackQuery) -> HandlerResult {
+    if let Ok(Some(State::ReceiveDateTime {
+        user_name,
+        service_name,
+    })) = dialogue.get().await
+    {
+        if let Some(time_slot) = q.data.as_ref() {
+            if let Some(message) = q.message.as_ref() {
+                let chat_id = message.chat.id;
+
+                bot.send_message(
+                    chat_id,
+                    format!(
+                        "😎Dear {user_name}❗\n
+                Your appointment for ➡️ {service_name} \n
+                at ➡️ {time_slot}' \n
+                ✅ Has been successfully booked!"
+                    ),
+                )
+                .reply_markup(InlineKeyboardMarkup::new(vec![
+                    vec![InlineKeyboardButton::callback("Yes, confirm", "confirm")],
+                    vec![InlineKeyboardButton::callback("No, change", "cancel")],
+                ]))
+                .await?;
+
+                dialogue
+                    .update(State::AppointmentConfirmed {
+                        user_name,
+                        service_name,
+                        appointment_time: time_slot.to_string(),
+                    })
+                    .await?;
+            } else {
+                log::error!("CallbackQuery without a message received: {:?}", q);
+            }
+        }
+    } else {
+        log::error!(
+            "Could not find information about your previous operation: {:?}",
+            q
+        );
+    }
+    Ok(())
+}
+
+async fn appointment_confirmation(
+    bot: Bot,
+    dialogue: MyDialogue,
+    q: CallbackQuery,
+    db_pool: web::Data<PgPool>,
+) -> HandlerResult {
+    if let Ok(Some(State::AppointmentConfirmed {
+        user_name,
+        service_name,
+        appointment_time,
+    })) = dialogue.get().await
+    {
+        save_appointment(db_pool, user_name, service_name, appointment_time)
+            .await
+            .expect("Ошибка сохранения записи");
```

Comment:
It's not a mistake. I think it is better to keep English messages everywhere.

Reply:
I agree, I didn't finish watching